### PR TITLE
Add more tutorials and add comments to existing ones

### PR DESCRIPTION
**Pull request by tristantarrant**
- Add jcache, distexec, remote, scripting, distributed, replicated, streams, query, listen tutorials
- Add comments to all tutorials
- Add ability to run each tutorial using mvn exec:exec

### CODE REVIEW

File: jcache/src/main/java/org/infinispan/tutorial/simple/jcache/InfinispanJCache.java [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package org.infinispan.tutorial.simple.jcache;
+
+import javax.cache.Cache;
+import javax.cache.CacheManager;
+import javax.cache.Caching;
+import javax.cache.configuration.MutableConfiguration;
+import javax.cache.spi.CachingProvider;
+
+public class InfinispanJCache {
```
Comment (galderz) [inline jcache/src/main/java/org/infinispan/tutorial/simple/jcache/InfinispanJCache.java]: I can expand it a bit further once this PR is in?

**Comment by galderz** [comment]
@tristantarrant I see there's a `listen/`tutorial for embedded listeners, and a `remote/` tutorial for remote. It'd probably make sense to add something on remote listeners? If so, where would it go? I assume in `listen/` ?

File: jcache/src/main/java/org/infinispan/tutorial/simple/jcache/InfinispanJCache.java (hunk 1 above)
Comment (tristantarrant) [inline jcache/src/main/java/org/infinispan/tutorial/simple/jcache/InfinispanJCache.java]: Yes. Let's keep it simple for now

**Comment by tristantarrant** [comment]
I 'd rather have one directory per example, so maybe remote-listen

**Comment by galderz** [comment]
@tristantarrant Sure, makes sense. I'll add one :). Integrating...

**Comment by galderz** [comment]
Integrated

**Pull request closed without merging**